src/pages/WorkloadList/FiltersAndSorts.ts

```diff
@@ -270,12 +269,8 @@ export namespace WorkloadListFilters {
     if (sortField.title === 'Health') {
       // In the case of health sorting, we may not have all health promises ready yet
       // So we need to get them all before actually sorting
-      const allHealthPromises: Promise<WorkloadItemHealth>[] = unsorted.map(item => {
-        return item.healthPromise.then(health => {
-          const withHealth: any = item;
-          withHealth.health = health;
-          return withHealth;
-        });
+      const allHealthPromises: Promise<WithHealth<WorkloadListItem>>[] = unsorted.map(item => {
+        return item.healthPromise.then((health): WithHealth<WorkloadListItem> => ({ ...item, ...{ health } }));
```

Comment:
isn't this the same as: `{ ...item, health }` ?? (just to have less nesting)

Reply:
Oh, that's a good one, will try. I tried `{ ...item, ...health }` but it didn't work, maybe this does.